compiler/rustc_resolve/src/late.rs

```diff
@@ -217,6 +217,8 @@ pub(crate) enum RibKind<'ra> {
     /// We passed through a `macro_rules!` statement
     MacroDefinition(DefId),
 
+    LookAheadMacroDefinition(DefId),
```

Comment:
The reason for using `RibKind::LookAheadMacroDefinition` instead of `RibKind::MacroDefinition` is to prevent cases like:

```rs
fn f() {
    let x = 0;
    macro_rules! foo { () => {
        assert_eq!(x, 0); // it will make `x` equal `1` if we use `RibKind::MacroDefinition`
    } }

    let x = 1;
    foo!();
}
```